### PR TITLE
User delete request

### PR DESCRIPTION
**Pull request by MichaelParadis**
Finished cancel request. We should be able to comment out the elastic search component when we add more elastic search functionality.

### CODE REVIEW

**Comment by MichaelParadis** [comment]
I pulled all the branches so here are my changes:
https://github.com/CMPUT301F16T01/Carrier/pull/73/commits/0e1822116cd80aa057d3b22aa88f082cffca74a3
https://github.com/CMPUT301F16T01/Carrier/pull/73/commits/8eecebb0e994d092f987165bcf870c825367a6b7
https://github.com/CMPUT301F16T01/Carrier/pull/73/commits/0d9fcf3c931a3bb4a48f9007b36a5e66a0fc957e
0b332329ed288f4cb9273d7ce01aad70b6b1d1bc
1d721c5ab5cf2856096869e2a4e996a48d3e2cdd
Sorry for making you go through individual commits.

**Comment by kieter** [comment]
Mandy is super busy with her interview and over the weekend so I'll be doing this review for you Mike.

File: app/src/main/java/comcmput301f16t01/github/carrier/ElasticRequestController.java
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/ElasticRequestController.java]: Push refactor pls. ( bad ) vs (good).

File: app/src/main/java/comcmput301f16t01/github/carrier/ElasticRequestController.java
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/ElasticRequestController.java]: Pretty please comment things. I'm not all that familiar with elastic search, I can kind of tell what's going on here, but it would be a lot more clear with some comments.

File: app/src/main/java/comcmput301f16t01/github/carrier/ElasticRequestController.java
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/ElasticRequestController.java]: I see that you are editing the status of a request to cancelled in elastic search upon successful retrieval of an existing request based on ID. I really think this needs some testing because as far as I can tell, I'm not sure if updating a request on elastic search to be cancelled is working. Maybe you could have put some placeholder requests in elastic search, cancelled them, and checked the status of them in elastic search to have  the courage that this works?
As far as I know, we don't have requests on elastic search (other than ones placed explicitly for testing), only user names. Updating elastic search was not really part of what you had to do (I think you were only intended to make gui work for cancelling and have the request list view update to cancelled for the request that was cancelled), I am glad that you started this though 👍 of course, you'll have to finish what you started.

File: app/src/main/java/comcmput301f16t01/github/carrier/RiderViewRequestActivity.java
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/RiderViewRequestActivity.java]: It is very useful that you initialized these, I just don't see why because you don't seem to use them here.

File: app/src/main/java/comcmput301f16t01/github/carrier/Location.java
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/Location.java]: Remove this comment, the TODO is implemented on line 33.

File: app/src/main/java/comcmput301f16t01/github/carrier/RequestController.java
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/RequestController.java:69]: Again, thanks for starting cancelling a request in elastic search, but please test this as I an unsure (an cannot check myself) whether or not this works.

Comment (MichaelParadis) [reply app/src/main/java/comcmput301f16t01/github/carrier/RequestController.java:69]: I commented this out in a newer commit because I was not testing it.

File: app/src/main/java/comcmput301f16t01/github/carrier/RiderViewRequestActivity.java
Comment (MichaelParadis) [reply app/src/main/java/comcmput301f16t01/github/carrier/RiderViewRequestActivity.java]: I use it in https://github.com/CMPUT301F16T01/Carrier/pull/73/commits/0d9fcf3c931a3bb4a48f9007b36a5e66a0fc957e
with setViewValues

**Review by kieter — CHANGES_REQUESTED** [review]
I am just generally going to briefly summarize my comments in the files. _I want you to read those._ What you were set out to do does not work, cancelling a request doesn't work as expected right now and I want this to be fixed. I am glad that you started changing the status of requests, but if you continue to work on this I want to see some tests and proof that it works. In general this pull request is really, really messy (I shouldn't have to search through all these commits to review your code) and I don't know why there are so many merge conflicts (did you pull everyone's branches?). The comments are sparse, and some code is confusing. This is a really good start, and I hope to see the final product 😄 

I won't merge this until all the merge conflicts are dealt with and your task is functional. Have this done by our next meeting (Wednesday, November 9) along with whatever task you are assigned this week.

File: app/src/main/java/comcmput301f16t01/github/carrier/RiderViewRequestActivity.java
Comment (kieter) [reply app/src/main/java/comcmput301f16t01/github/carrier/RiderViewRequestActivity.java]: Ya haha I saw that. I was gonna delete this comment, I just couldn't find it.

**Comment by kieter** [comment]
Btw I tested this by checking out your branch and running the app from there.

**Comment by kieter** [comment]
Jk do the following:
- [ ] Update adapter
- [x] Go back after cancel
- [x] Fix merge conflicts
- [x] Comment code
- [x] JavaDoc code
- [ ] Write tests

File: app/src/main/java/comcmput301f16t01/github/carrier/RequestController.java
Comment (MichaelParadis) [inline app/src/main/java/comcmput301f16t01/github/carrier/RequestController.java:58]: This line is causing your problems future mike.

**Comment by kieter** [comment]
How's this pull request going?

**Comment by MichaelParadis** [comment]
Having trouble updating the adapter and I had to remake the branch because of some mistakes. I will try and figure out the adapter later tonight but everything else should be addressed. I will remake the pull request later today.

**Comment by kieter** [comment]
Closed because Mike made a new branch with this stuff in it because there were so many conflicts.

**Pull request closed without merging**
